.github/workflows/ccpp.yml

```diff
@@ -3,15 +3,116 @@ name: C/C++ CI
 on: [push, pull_request]
 
 jobs:
-  build:
+  unit_and_style:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: build
+      run:  scripts/build_ci.bash
+
+    - name: Check style
+      run:  make compliant && make shellcheck && make docs-coverage
+
+    - name: Check install
+      run:  sudo make install && make install-check
+
+    - name: Run unit tests
+      run:  make unit-check
+
+    - name: Print status
+      if: failure()
+      run: cat test-suite.log
+
+  group1:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: build
+      run:  scripts/build_ci.bash
+
+    - name: Run Check
+      run: env TESTS="$(find test/functional/update -name *.bats -printf '%p ')" make -e -j2 check
+
+    - name: Print status
+      if: failure()
+      run: cat test-suite.log
+
+  group2:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: build
+      run:  scripts/build_ci.bash
+
+    - name: Run Check
+      run: env TESTS="$(find test/functional/{autoupdate,bundleinfo,verify-legacy,checkupdate,hashdump,mirror,usability} -name *.bats \( ! -name usa-config-file.bats \) -printf '%p ')" make -e -j2 check
+
+    - name: Run Check - config file
+      run: env TESTS="$(find test/functional/usability -name usa-config-file.bats -printf '%p ')" make -e -j2 check
+
+    - name: Print status
+      if: failure()
+      run: cat test-suite.log
 
+  group3:
     runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: build
+      run:  scripts/build_ci.bash
 
+    - name: run check
+      run: env tests="$(find test/functional/{diagnose,search} -name *.bats -printf '%p ')" make -e -j2 check
```

Comment:
I think I know why your jobs are taking this long... your variable is called `tests` and I think it has to be called `TESTS`, so make check is ignoring it therefore running all tests in the job, not just the one specified in tests.